Implement model v1.1

janosbrodbeck opened the pull request.

dolevi-tub commented on src/app/schemas/robot_type.py:
```diff
@@ -3,25 +3,34 @@
 class RobotTypeUpdate(RobotTypeBase):
-    name: str
-    vendor: str
+    name: Optional[str] = None
+    vendor: Optional[str] = None
```
This means an update with body `{ "name": "NiryoOne" }` will leave `vendor` as is (and don't set it to `None`), correctly?

janosbrodbeck replied:
Should not change it, yes. Either way, it was a leftover and not intended. Thanks for the catch! Fix is already squashed in.

janosbrodbeck replied:
Should not change it, yes. Either way, it was a leftover and not intended. Thanks for the catch! Fix is already squashed in.

janosbrodbeck commented:
Also added the endpoint to update multiple welding points at once. In this case we require, that every element sent to update contains its ID. Added a check before to ensure that

janosbrodbeck commented on src/app/api/api_v1/endpoints/welding_point.py:
```diff
@@ -0,0 +1,98 @@
+    # Check, if each element has a proper ID
+    if not all(wp.id is not None for wp in welding_points_in):
+        raise HTTPException(status_code=400, detail="To update multiple welding points each element needs to contain "
+                                                    "its ID")
```
Also simplified the check expression for this condition

janosbrodbeck commented on src/app/api/api_v1/endpoints/welding_point.py:
```diff
@@ -0,0 +1,98 @@
+    """
+    Update a welding point
+    """
+    result = await welding_point.get_by_id(db=db, id=id)
```
Addressed, same for the others. Not really liking it, that we have `_id` for the endpoints, but at least we don't have the shadow warning anymore.

dolevi-tub reviewed (approved):
All okay now!
